File: EBNF/GangOfFour/Java/OOPPatternsJavaApp/src/AdapterPattern/ClassAFormat1.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package AdapterPattern;

/**
 *
 * @author Vlado
 */
public class ClassAFormat1 implements IStringProvider {
    private ClassA classA = null;
    
    public ClassAFormat1(final ClassA a)
    {
        classA = a;
    }
    
    public String getStringData() {
        return format(classA.getStringData());
    }

    private String format(final String sourceValue) {
        // Manipulate the source string into a format required 
        // by the object needing the source object's data
        return sourceValue.trim();
    }
    
    /*public String getStringData()
    {
        String ishod = "";
        
        return ishod;
    }*/
}

File: EBNF/GangOfFour/CSharp/OOPPatternsWpf/OOPPatternsWpf/AdapterTrial3/ClassAFormat1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OOPPatternsWpf.AdapterTrial3
{
    class ClassAFormat1:IStringProvider
    {
        public ClassA classAObject;

        public ClassAFormat1(ClassA obj)
        {
            classAObject = obj;
        }

        public String getStringData()
        {
            return format(classAObject.getStringData());
        }

        private string format(string sourceValue)
        {
            // Manipulate the source string into a format required 
            // by the object needing the source object's data
            return "----------------" + sourceValue.Trim() + " ------------- ";
        }
    }
}
